google/cloud/sql/connector/instance_connection_manager.py

```diff
@@ -350,6 +357,27 @@ def _auth_init(self) -> None:
 
         self._credentials = credentials
 
+    async def _force_refresh(self) -> bool:
+        if self._refresh_in_progress.is_set():
+            # if a new refresh is already in progress, then block on the result
+            self._current = await self._next
+            return True
+        try:
+            self._next.cancel()
+            # schedule a refresh immediately with no delay
+            self._next = self._loop.create_task(self._schedule_refresh(0))
+            self._current = await self._next
+            return True
+        except Exception as e:
+            # if anything else goes wrong, log the error and return false
+            logger.exception("Error occurred during force refresh attempt", exc_info=e)
+            return False
+
+    def force_refresh(self, timeout: Optional[int] = None) -> bool:
```

Comment:
nit: This function should probably have a docstring

Reply:
I think this would still be a good idea

Reply:
added a docstring